packages/authentication/src/services/token.service.ts

```diff
@@ -11,10 +11,32 @@ import {UserProfile} from '@loopback/security';
 export interface TokenService {
   /**
    * Verifies the validity of a token string and returns a user profile
+   *
+   * @param token The token/secret which should be validated/verified.
+   * @param userProfile Depending on the token system this optional parameter
+   *   can be used to check if the token is valid for the given user.
+   *
+   * @returns The UserProfile which belongs to the given token.
    */
-  verifyToken(token: string): Promise<UserProfile>;
+  verifyToken(token: string, userProfile?: UserProfile): Promise<UserProfile>;
+
   /**
    * Generates a token string based on a user profile
+   *
+   * @param userProfile A UserProfile for which a token should be generated.
+   *
+   * @returns a generated token/secret for a given UserProfile.
    */
   generateToken(userProfile: UserProfile): Promise<string>;
+
+  /**
+   * Revokes a given token (if supported by token system)
+   *
+   * @param token The token/secret which should be revoked/invalidated.
+   * @param userProfile An optional UserProfile to check if the token
+   *   is valid for a given User.
+   *
+   * @returns true, if the given token was invalidated.
+   */
+  revokeToken(token: string, userProfile?: UserProfile): Promise<boolean>;
```

Comment:
What about make it optional? `revokeToken?(...)`?

Reply:
Thanks for your feedback. I didn't think of this simple solution - this makes the change non-breaking, but more complex in using it. How you would check for existance of this method?

I'm also not sure if it's better to return an boolean or just throw an exception if a token can not be invalidated. In the light of making revokeToken optional, do you suggest to just throw?

Reply:
I was wondering about that too. I guess `Promise<boolean>` is fine for now. If `revokeToken` is not present, we can default to `async() => false`.

Reply:
Changed to be optional.